consumer.robot

```diff
@@ -35,33 +36,54 @@ Login
         Fail    Login failed
     END
 
+Return to start
+    [Documentation]
+    ...    Simulates reseting the robot's environment to be ready for the next work item.
+    ...    Actual code could include navigating to a home page.
+    Log    Returning to start position...
+
 Action for item
     [Documentation]
-    ...    Simulates handling of one work item that fails 1/5 times to
-    ...    highlight BUSINESS exception handling.
+    ...    Simulates handling of one work item that fails 1/5 times in
+    ...    two different ways to highlight BUSINESS and APPLICATION exception handling
+    ...    and how you can handle each differently from the main "Handle item" keyword.
     [Arguments]    ${payload}
-    ${Item Action OK}=    Evaluate    random.randint(1, 5)
-    IF    ${Item Action OK} != 1
-        Log    Did a thing item for: ${payload}
-    ELSE
-        Fail    Failed to handle item for: ${payload}.
+    ${Item Action OK}=    Evaluate    random.randint(1, 10)
+    IF    ${Item Action OK} not in [1,2]
+        Log    Did the first thing for: ${payload}
+    ELSE IF    ${Item Action OK} == 1
+        Fail    Invalid data in payload: ${payload}.
+    ELSE IF    ${Item Action OK} == 2
+        Fail    Application timed out, try again later.
     END
 
 Handle item
     [Documentation]    Error handling around one work item.
-    ${payload}=    Get Work Item Payload
-    ${Item Handled}=    Run Keyword And Return Status
-    ...    Action for item    ${payload}
-    IF    ${Item Handled}
-        Release Input Work Item    DONE
-    ELSE
+    ${payload}=    Get Work Item Variables
+    TRY
```

Comment:
This TRY-EXCEPT block is good for detecting specific errors but if any other error than expected 2 cased happens then those are not caught and this block will fail to handle that.